internal/scheduler/reports.go

```diff
@@ -330,42 +287,147 @@ func (repo *SchedulingContextRepository) GetSchedulingReport(_ context.Context,
+			fmt.Fprint(w, indent.String("\t", "Most recent scheduling round that scheduled a job:\n"))
 			fmt.Fprint(w, indent.String("\t\t", sctx.ReportString(verbosity)))
 		} else {
-			fmt.Fprint(w, indent.String("\t", "Most recent successful attempt: none\n"))
+			fmt.Fprint(w, indent.String("\t", "Most recent scheduling round that scheduled a job: none\n"))
 		}
-		if sctx := sr.mostRecentPreemptingByExecutor[executorId]; sctx != nil {
-			fmt.Fprint(w, indent.String("\t", "Most recent preempting attempt:\n"))
+		if sctx := mostRecentPreemptingByExecutor[executorId]; sctx != nil {
+			fmt.Fprint(w, indent.String("\t", "Most recent scheduling round that preempted a job:\n"))
 			fmt.Fprint(w, indent.String("\t\t", sctx.ReportString(verbosity)))
 		} else {
-			fmt.Fprint(w, indent.String("\t", "Most recent preempting attempt: none\n"))
+			fmt.Fprint(w, indent.String("\t", "Most recent scheduling round that preempted a job: none\n"))
+		}
+	}
+	w.Flush()
+	return sb.String()
+}
+
+func (repo *SchedulingContextRepository) getSchedulingReportStringForQueue(queue string, verbosity int32) string {
+	mostRecentByExecutor, _ := repo.GetMostRecentSchedulingContextByExecutorForQueue(queue)
+	mostRecentSuccessfulByExecutor, _ := repo.GetMostRecentSuccessfulSchedulingContextByExecutorForQueue(queue)
+	mostRecentPreemptingByExecutor, _ := repo.GetMostRecentPreemptingSchedulingContextByExecutorForQueue(queue)
+	var sb strings.Builder
+	w := tabwriter.NewWriter(&sb, 1, 1, 1, ' ', 0)
+	for _, executorId := range repo.GetSortedExecutorIds() {
+		fmt.Fprintf(w, "%s:\n", executorId)
+		if sctx := mostRecentByExecutor[executorId]; sctx != nil {
+			fmt.Fprintf(w, "\tMost recent scheduling round that considered queue %s:\n", queue)
+			sr := getSchedulingReportForQueue(sctx, queue)
+			fmt.Fprint(w, indent.String("\t\t", sr.ReportString(verbosity)))
+		} else {
+			fmt.Fprintf(w, "\tMost recent scheduling round that considered queue %s: none\n", queue)
+		}
+		if sctx := mostRecentSuccessfulByExecutor[executorId]; sctx != nil {
+			fmt.Fprintf(w, "\tMost recent scheduling round that scheduled a job from queue %s:\n", queue)
+			sr := getSchedulingReportForQueue(sctx, queue)
+			fmt.Fprint(w, indent.String("\t\t", sr.ReportString(verbosity)))
+		} else {
+			fmt.Fprintf(w, "\tMost recent scheduling round that scheduled a job from queue %s: none\n", queue)
+		}
+		if sctx := mostRecentPreemptingByExecutor[executorId]; sctx != nil {
+			fmt.Fprintf(w, "\tMost recent scheduling round that preempted a job from queue %s:\n", queue)
+			sr := getSchedulingReportForQueue(sctx, queue)
+			fmt.Fprint(w, indent.String("\t\t", sr.ReportString(verbosity)))
+		} else {
+			fmt.Fprintf(w, "\tMost recent scheduling round that preempted a job from queue %s: none\n", queue)
+		}
+	}
+	w.Flush()
+	return sb.String()
+}
+
+func (repo *SchedulingContextRepository) getSchedulingReportStringForJob(jobId string, verbosity int32) string {
```

Comment:
This function currently only uses `repo.mostRecentByExecutorByJobId`, which may be out of sync with the other parts of `repo` (e.g., it may not contain all of the jobs from the most recent scheduling round for each queue); we should consider falling back on those other maps.